editor/src/components/canvas/canvas-utils.spec.browser.ts

```diff
@@ -1401,6 +1404,336 @@ describe('moveTemplate', () => {
     )
   })
 
+  it('reparents an orphan from the canvas', async () => {
+    const renderResult = await renderTestEditorWithCode(
+      Prettier.format(
+        `/** @jsx jsx */
+      import * as React from 'react'
+      import { Scene, Storyboard, View, jsx } from 'utopia-api'
+    
+      export var App = (props) => {
+        return (
+          <div style={{ position: 'relative', width: '100%', height: '100%'}} data-uid='aaa' />
+        )
+      }
+    
+      export var ${BakedInStoryboardVariableName} = (props) => {
+        return (
+          <Storyboard data-uid='${BakedInStoryboardUID}'>
+            <Scene
+              style={{ position: 'absolute', left: 0, top: 0, width: 400, height: 400 }}
+              component={App}
+              props={{}}
+              data-uid='scene-aaa'
+            />
+            <div
+              style={{
+                backgroundColor: '#0091FFAA',
+                position: 'absolute',
+                width: 100,
+                height: 100,
+                left: 350,
+                top: 0,
+              }}
+              data-uid='orphan-bbb'
+              data-testid='orphan-bbb'
+            />
+          </Storyboard>
+        )
+      }`,
+        PrettierConfig,
+      ),
+    )
+
+    await renderResult.dispatch(
+      [
+        selectComponents(
+          [TP.instancePath(TP.scenePath([]), [BakedInStoryboardUID, 'orphan-bbb'])],
+          false,
+        ),
+      ],
+      false,
+    )
+
+    const areaControl = renderResult.renderedDOM.getByTestId('orphan-bbb')
+
+    const areaControlBounds = areaControl.getBoundingClientRect()
+
+    const canvasControlsLayer = renderResult.renderedDOM.getByTestId(CanvasControlsContainerID)
+
+    fireEvent(
+      canvasControlsLayer,
+      new MouseEvent('mousedown', {
+        bubbles: true,
+        cancelable: true,
+        metaKey: true,
+        clientX: areaControlBounds.left + 5,
+        clientY: areaControlBounds.top + 5,
+        buttons: 1,
+      }),
+    )
+
+    await act(async () => {
+      const domFinished = renderResult.getDomReportDispatched()
+      const dispatchDone = renderResult.getDispatchFollowUpactionsFinished()
+      fireEvent(
+        canvasControlsLayer,
+        new MouseEvent('mousemove', {
+          bubbles: true,
+          cancelable: true,
+          metaKey: true,
+          clientX: areaControlBounds.left - 5,
+          clientY: areaControlBounds.top + 5,
+          buttons: 1,
+        }),
+      )
+      await domFinished
+      await dispatchDone
+    })
+
+    await act(async () => {
```

Comment:
is this the same mouse move event as above? do we need two of these?

Reply:
I've seen this for other reparent tests, when I tried it with only one mousemove event the reparent is not happening and the element stays where it was.

Reply:
okay :)